packages/venia-concept/src/i18n.js

```diff
@@ -0,0 +1,34 @@
+import i18n from 'i18next';
+import merge from 'lodash/merge';
+import resourcesVeniaUI from '@alienfast/i18next-loader!@magento/venia-ui/lib/locales/index.js';
+import resourcesLocal from '@alienfast/i18next-loader!./locales/index.js';
+import { Util } from '@magento/peregrine';
+const { BrowserPersistence } = Util;
+
+/**
+ * Initialize the i18n object for our app
+ * Also loads our translation files
+ */
+export const initi18n = () => {
+    /**
+     * Merge Venia UI and Local Resources together
+     * We can add additional resources here (maybe from peregrine for example)
+     * Would want to look at exposing this to the extensibility work also
+     */
+    const resources = merge(resourcesVeniaUI, resourcesLocal);
```

Comment:
Yes! We absolutely want to merge and override here. Seems like the pattern here is already most of the way there; you've got a loader which renders this data as JSON, and you could change it to codegen which yields the JSON directly. To match a pattern which we've used elsewhere:

1.  Replace the two calls to Venia and local with a single call: `import resources from '@absolunet/pwa-i18n/magic-resources'` or something.
2. Implement `magic-resources` to assemble the localization files in all dependencies
3. Use the new `transformModules` Target to apply a [`val-loader`](https://webpack.js.org/loaders/val-loader/) to that file only

You could call your own targets in there, exposing your own localization extension points.
4.